src/HttpClientRequestAdapter.cs

```diff
@@ -47,6 +47,7 @@ public HttpClientRequestAdapter(IAuthenticationProvider authenticationProvider,
             authProvider = authenticationProvider ?? throw new ArgumentNullException(nameof(authenticationProvider));
             createdClient = httpClient == null;
             client = httpClient ?? KiotaClientFactory.Create();
+            BaseUrl = client.BaseAddress.ToString();

```

Comment:
Can we add a check if the `BaseAddress` is null? As the property is nullable.
https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.baseaddress?view=net-7.0